xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -37,135 +40,129 @@ CEdl::CEdl()
-      !URIUtils::IsInternetStream(strMovie))
+  bool found = false;
+
+  // Only check for edit decision lists if the movie is on the local hard drive, or accessed over a
+  // network share.
+  const std::string& path = fileItem.GetDynPath();
+  if ((URIUtils::IsHD(path) || URIUtils::IsOnLAN(path)) && !URIUtils::IsInternetStream(path))
   {
-    CLog::Log(LOGDEBUG,
-              "{} - Checking for edit decision lists (EDL) on local drive or remote share for: {}",
-              __FUNCTION__, CURL::GetRedacted(strMovie));
+    CLog::LogF(LOGDEBUG,
+               "Checking for edit decision lists (EDL) on local drive or remote share for: {}",
+               CURL::GetRedacted(path));
 
-    /*
-     * Read any available file format until a valid EDL related file is found.
-     */
-    if (!bFound)
-      bFound = ReadVideoReDo(strMovie);
+    // Read any available file format until a valid EDL related file is found.
+    if (!found)
+      found = ReadVideoReDo(path);
 
-    if (!bFound)
-      bFound = ReadEdl(strMovie, fFramesPerSecond);
+    if (!found)
+      found = ReadEdl(path, fps);
 
-    if (!bFound)
-      bFound = ReadComskip(strMovie, fFramesPerSecond);
+    if (!found)
+      found = ReadComskip(path, fps);
 
-    if (!bFound)
-      bFound = ReadBeyondTV(strMovie);
+    if (!found)
+      found = ReadBeyondTV(path);
   }
   else
   {
-    bFound = ReadPvr(fileItem);
+    found = ReadPvr(fileItem);
   }
 
-  if (bFound)
+  if (found)
   {
     MergeShortCommBreaks();
     AddSceneMarkersAtStartAndEndOfEdits();
   }
 
-  return bFound;
+  return found;
 }
 
-bool CEdl::ReadEdl(const std::string& strMovie, const float fFramesPerSecond)
+bool CEdl::ReadEdl(const std::string& path, float fps)
 {
   Clear();
 
-  std::string edlFilename(URIUtils::ReplaceExtension(strMovie, ".edl"));
+  std::string edlFilename{URIUtils::ReplaceExtension(path, ".edl")};
   if (!CFile::Exists(edlFilename))
     return false;
 
   CFile edlFile;
   if (!edlFile.Open(edlFilename))
   {
-    CLog::Log(LOGERROR, "{} - Could not open EDL file: {}", __FUNCTION__,
-              CURL::GetRedacted(edlFilename));
+    CLog::LogF(LOGERROR, "Could not open EDL file: {}", CURL::GetRedacted(edlFilename));
     return false;
   }
 
-  bool bError = false;
-  int iLine = 0;
-  std::string strBuffer;
-  strBuffer.resize(1024);
-  while (edlFile.ReadString(&strBuffer[0], 1024))
+  bool error{false};
+  int line{0};
+  std::array<char, 1024> buffer{};
+  while (edlFile.ReadString(&buffer[0], 1024))
   {
     // Log any errors from previous run in the loop
-    if (bError)
-      CLog::Log(LOGWARNING, "{} - Error on line {} in EDL file: {}", __FUNCTION__, iLine,
-                CURL::GetRedacted(edlFilename));
+    if (error)
+      CLog::LogF(LOGWARNING, "Error on line {} in EDL file: {}", line,
+                 CURL::GetRedacted(edlFilename));
 
-    bError = false;
+    error = false;
+    line++;
 
-    iLine++;
+    std::array<char, 513> buffer1{};
```

Comment:
Do you know where this odd size comes from?

Reply:
I think the reason is:
we want to get/break two buffers from the 1024 buffer, hence 512+null terminator = 513

This whole logic and the usage of sscanf should be replaced. As far as I could find, there's not a clean equivalent for c++. We also have limitations on the IFile implementation